setup.py

```diff
@@ -262,6 +262,11 @@ def finalize_options(self):
             include_dirs=[model_dir]),
         Extension('gensim._matutils',
             sources=['./gensim/_matutils.c']),
+        Extension('gensim.models.sent2vec_inner',
+            sources=['./gensim/models/sent2vec_inner.cpp'],
+            include_dirs=[model_dir],
+            language="c++",
+            extra_compile_args=['/std:[c++14|c++latest]', '/EHsc'])
```

Comment:
remember that you need different arguments for different OS version here
- `-std=c++11` for non-windows
- `something-else` for windows

You can make something like

`extra_compile_args=['/std:[c++14|c++latest]' if os.name == "nt" else "-std=c++11"]`

for each argument that is OS-special